src/Security/Security.php

```diff
@@ -860,14 +943,34 @@ public function changepassword()
     }
 
     /**
+     * Create a link to the password reset form.
+     *
+     * GET parameters used:
+     * - m: member ID
+     * - t: plaintext token
+     *
+     * @param Member $member Member object associated with this link.
+     * @param string $autologinToken The auto login token.
+     * @return string
+     */
+    public static function getPasswordResetLink($member, $autologinToken)
+    {
+        $autologinToken = urldecode($autologinToken);
+        $selfControllerClass = __CLASS__;
+        /** @var static $selfController */
+        $selfController = new $selfControllerClass();
+        return $selfController->Link('changepassword') . "?m={$member->ID}&t=$autologinToken";
```

Comment:
return static::singleton()->Link('changepassword') . "?m={$member->ID}&t=$autologinToken";

Save from creating new class instances, re-use injected dependencies, etc. :)
@tractorcow
tractorcow 2 days ago Contributor

Alternatively, make this method non-static.

static::singleton()->getPasswordResetLink($member, $autologinToken);